datumaro/datumaro/util/image.py

```diff
@@ -45,7 +46,17 @@ def load_image(path):
         assert image.shape[2] in {3, 4}
     return image
 
-def save_image(path, image, **kwargs):
+def save_image(path, image, create_dir=False, **kwargs):
+    # NOTE: Check destination path for existence
+    # OpenCV silently fails if target directory does not exist
+    dst_dir = osp.dirname(path)
+    if dst_dir:
+        if create_dir:
+            os.makedirs(dst_dir, exist_ok=True)
+        elif not osp.isdir(dst_dir):
+            raise NotADirectoryError(
```

Comment:
FileNotFoundError - Raised when a file or directory is requested but doesn’t exist. Corresponds to errno ENOENT.